### PR TITLE
Refactor getting the index of a user's language 

### PR DESCRIPTION
**Pull request by davidculley**
- Simplified the for-loop within a for-loop.
- Self-documented the code.
- Fixed a few type checker warnings.

### CODE REVIEW

**Comment by dae** [comment]
Hi David,

Thank you for all the time you've put into your recent PRs - your enthusiasm is clearly visible!

Quite a few of your recent PRs have been focused on refactoring - a bit too much I'm afraid. Every PR has a cost to review, and carries a risk of accidentally introducing a bug/regression. Trivial changes that are clearly correct get some leeway, but I've been bitten in the past by not noticing problems with "obviously correct" code :-)

Instead, what I'd recommend doing is trying to limit refactoring to when you're working on that part of the code anyway (e.g. to fix a bug or add a feature), or when there's a compelling reason to make the change.

Please note this is not a reflection on the quality of your changes, or an argument that the original code is better. I just think when we know the existing code works, we should avoid modifying it unless the benefits outweigh the risks.

My apologies that I didn't find time to make this clearer to you sooner. :-(

**Comment by dae** [comment]
On the topic of useful refactors, I've just created a new issue to document a refactoring that I do think would be worth it, in case you're interested: https://github.com/ankitects/anki/issues/3317

File: pylib/anki/lang.py [hunk 1]
```diff
@@ -180,9 +180,17 @@ def set_lang(lang: str) -> None:
...
+def get_def_lang(user_lang: str | None = None) -> tuple[int, str]:
+    """Return user_lang converted to name used on disk and its index, defaulting to system language
     or English if not available."""
+
+    def get_index_of_language(wanted_locale: str) -> int | None:
+        indices_generator = (
+            i for i, (_, locale_) in enumerate(langs) if locale_ == wanted_locale
+        )
+        index = next(indices_generator, None)
+        return index
```
[the remark this answers is not in the document]
Comment (davidculley) [reply pylib/anki/lang.py]: Fixed (1e43e9d).

[the remark this answers is not in the document]
Comment (dae) [reply pylib/anki/lang.py]: Thanks!

**Comment by davidculley** [comment]
> I've been bitten in the past by not noticing problems with "obviously correct" code :-)

Totally relatable.

> Thank you for all the time you've put into your recent PRs - your enthusiasm is clearly visible!

Thank you for recognizing my efforts!

> Every PR has a cost to review, and carries a risk of accidentally introducing a bug/regression.

Totally understandable.

> Please note this is not a reflection on the quality of your changes, or an argument that the original code is better. 

It's all good, don't fret. 🙂

**Pull request merged**